### PR TITLE
OAK-8995: remove support for lucene custom scorer

### PR DESCRIPTION
**Pull request by fabriziofortino**
https://issues.apache.org/jira/browse/OAK-8995

### CODE REVIEW

**Review by reschke — COMMENTED** [review]
I find it problematic that this change mixes cleanup (Java 8, typo fixes etc) with actual changes. These really should be seperated.

That is, I would pick just the first commit, and do the other changes in a separate issue.

**Comment by fabriziofortino** [comment]
@reschke I see your point. The different changes are actually in multiple commits and can be seen separately. Creating a separate OAK issue and PR would surely be cleaner but is not slimline in terms of process, especially taking into account the fact we have to merge it back to svn. IMHO this discourages continuous improvements. Limiting these improvements only to the files actually changed and separating them in different commits seems a good tradeoff.

**Comment by reschke** [comment]
Well, there's always a tradeoff. The purpose of separating unrelated changes is (a) so they are easier to review, and (b) it's easier to backport them to older releases. Making things simpler for *this* change might make it harder to backport, for instance because the change does not apply cleanly anymore for the simple reason it is larger.

File: oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexTest.java [hunk 1]
```diff
@@ -980,26 +969,6 @@ private void assertQuery(IndexTracker tracker, NodeState indexed, String key, St
         return paths;
     }
 
-    private static List<String> assertFilter(Filter filter, AdvancedQueryIndex queryIndex,
```
Comment (nit0906) [inline oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexTest.java:1058]: Was this removal by mistake ? I see it's still called in this test class.

**Review by nit0906 — COMMENTED** [review]
LGTM, just one issue in a test class that seems like a problem.

File: oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexTest.java (hunk 1 above)
Comment (fabriziofortino) [reply oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexTest.java:1058]: This one is not used at all. The one in use does not specify the `ordered` boolean parameter (see line 954).

Comment (nit0906) [reply oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexTest.java:1058]: ah ..ok now I see (I was checking in trunk where it is called from customScoreQuery(), but even that method itself is removed in your changes.)

**Review by nit0906 — APPROVED** [review]
LGTM

**Review by thomasmueller — COMMENTED** [review]
Yes, the patch contains unrelated changes... So far we have preferred having separate commits for unrelated changes, for example to simplify backports (if needed). But I'm fine with keeping them here.

**Review by thomasmueller — APPROVED** [review]
I don't think we will ever backport this, so I think the current changes are fine.

**Comment by fabriziofortino** [comment]
merged with https://issues.apache.org/jira/browse/OAK-8995

**Pull request closed without merging**